### PR TITLE
New pre-commit commands inc isort and auto ipynb example conversion

### PR DESCRIPTION
**Pull request by je-cook**
## Description

Adds some more pre-commit commands. isort I've been slowly running and changing bits of code as I go for a while. I think its a nice cleanup and should remove any import based merge conflicts when rebasing (from now on). 

I've also added an auto-conversion for example scripts to notebook files. There are some that do not yet have comments (run `SKIP=isort,black,flake8,black-jupyter pre-commit run --all-files` to see the examples I havent fixed).

Finally there are some file encoding auto-fixes from pre-commit itself (this is the bulk of the file changes).

I would review this per commit as I have tried to contain the changes into reasonable chunks.

## Interface Changes

N/A

## Checklist

I confirm that I have completed the following checks:

- [X] Tests run locally and pass `pytest tests --reactor`
- [X] Code quality checks run locally and pass `flake8` and `black .`
- [X] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by je-cook** [comment]
the last 7 commits are automatic changes so I will update them whenever this is ready

File: examples/BLUEPRINT/base/base_tutorial.ipynb [hunk 1]
```diff
@@ -49,7 +46,7 @@
         "    default_params = [\n",
         "        [\"n_TF\", \"Number of TF coils\", 16, \"N/A\", None, \"Input\"],\n",
         "        [\"coolant\", \"Coolant\", \"Water\", None, \"Divertor coolant type\", \"Common sense\"],\n",
-        "        [\"T_in\", \"Coolant inlet T\", 80.0, \"°C\", \"Coolant inlet T\", None],\n",
+        "        [\"T_in\", \"Coolant inlet T\", 80.0, \"\u00b0C\", \"Coolant inlet T\", None],\n",
```
Comment (DanShort12) [inline examples/BLUEPRINT/base/base_tutorial.ipynb:49]: Any reason for this change? (I guess it relates to the on-going UTF-8 vs ascii issues).

**Review by DanShort12 — COMMENTED** [review]
A nice addition for keeping things consistent. Just the one comment where we seem to have converted some UTF-8 to ascii. I know there are on-going issues for some people with non-ascii characters but would be good to understand if this is related or if one of the hooks is doing that conversion and would convert UTF-8 characters even when the encoding issues are resolved.

File: examples/BLUEPRINT/base/base_tutorial.ipynb (hunk 1 above)
Comment (je-cook) [reply examples/BLUEPRINT/base/base_tutorial.ipynb:49]: I think possibly @CoronelBuendia changed it in the py file? I remember seeing it and assuming the same...

Comment (DanShort12) [reply examples/BLUEPRINT/base/base_tutorial.ipynb:49]: Ah, fair enough, makes sense that the notebook aligns with the .py then.

**Review by DanShort12 — APPROVED** [review]
Looks fine now, just needs a rebase.

**Pull request merged**